synapse/config/saml2_config.py

```diff
@@ -226,6 +234,11 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
           #
           #mxid_mapping: dotreplace
 
+          # A third-party module can be provided here as a custom solution to mapping the
+          # above configured saml attribute onto a matrix ID. If this is defined, it will
+          # override the `mxid_mapping` option.
+          #mxid_mapping_provider: "mapping_provider.SamlMappingProvider"
```

Comment:
also, we should follow the conventions for modules elsewhere of having a dict with `module` and `config` keys.